Source/WebCore/dom/Element+CSSOMView.idl

```diff
@@ -27,6 +27,9 @@
 partial interface Element {
     DOMRectList getClientRects();
     [NewObject] DOMRect getBoundingClientRect();
+
+    boolean checkVisibility(optional CheckVisibilityOptions options = {});
```

Comment:
We should puts this under a runtime flag at least.

Reply:
I can add a flag to disable, but we'll want it on by default so reader can make use of this API.

Reply:
We have the ability to expose API to a specific DOM wrapper world. See `Element::shadowRootForBindings` and `InjectedBundleScriptWorld::makeAllShadowRootsOpen()` and `collectMatchingElementsInFlatTree` and `matchingElementInFlatTree` in `LocalDOMWindow.idl`.